data/World/Overworld.json

```
@@ -1386,7 +1386,18 @@
         "region_name": "Market Treasure Chest Game",
         "scene": "Market Treasure Chest Game",
         "locations": {
-            "Market Treasure Chest Game Reward": "can_use(Lens_of_Truth)"
+            "Market Treasure Chest Game Salesman": "True",
+            "Market Treasure Chest Game Reward": "(shuffle_tcgkeys != 'vanilla' or can_use(Lens_of_Truth) and (Small_Key_Treasure_Chest_Game, 6))",
+            "Market Treasure Chest Game Room 1 Bottom": "(shuffle_tcgkeys != 'vanilla' and (Small_Key_Treasure_Chest_Game, 1))",
```

```suggestion
            "Market Treasure Chest Game Room 1 Bottom": "(Small_Key_Treasure_Chest_Game, 1)",
```

Similar for the other 9 locations.